src/Orleans/Facet/FacetAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿
+using System;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Marker attribute for facets
+    /// </summary>
+    public abstract class FacetAttribute : Attribute
```

Comment:
Regarding catching errors, it's not very different if we don't have a mapper registered, since you'll get a clear error if you try to resolve `IExampleStorage<T>` because there will be no service registered for it.

Regarding perf, it doesn't need to consider all attributes, just attributes that have a mapper associated with it. Also the (very minimal) cost is only paid once anyway, since reflection only happens the first time the grain type is encountered for activation. Also, even not having abstract attributes, and just have sealed ones is better for perf, so we "pay" for checking for a few attributes, but with faster look ups each. Having said that, I wouldn't put perf as a pro or con to keep the base attribute, as there is no difference after the type is cached.

I kind of like the idea of being able to use non-Orleans attributes, or even reuse `StorageProviderAttribute` for example, so you can use the old and the new approach with the same attribute, but varies whether you apply it to the `Grain<T>` class, or to a constructor parameter argument (or to an injected property in the future). After all it's just metadata, that can be interpreted by different logic in different contexts, to get the same behavior.

@sergeybykov, @ReubenBond, ??, thoughts?